@here/olp-sdk-dataservice-read/test/unit/VersionedLayerClientNew.test.ts

```diff
@@ -0,0 +1,666 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import * as chai from "chai";
+import sinon = require("sinon");
+import sinonChai = require("sinon-chai");
+
+import { BlobApi, MetadataApi, QueryApi } from "@here/olp-sdk-dataservice-api";
+import { Index } from "@here/olp-sdk-dataservice-api/lib/query-api";
+import * as dataServiceRead from "../../lib";
+
+chai.use(sinonChai);
+
+const assert = chai.assert;
+const expect = chai.expect;
+
+// VersionedLayerClient test with new constructor.
+describe("VersionedLayerClientNew", () => {
+    let sandbox: sinon.SinonSandbox;
+    let getBlobStub: sinon.SinonStub;
+    let getPartitionsStub: sinon.SinonStub;
+    let getPartitionsByIdStub: sinon.SinonStub;
+    let getQuadTreeIndexStub: sinon.SinonStub;
+    let getVersionStub: sinon.SinonStub;
+    let getBaseUrlRequestStub: sinon.SinonStub;
+    let versionedLayerClientNew: dataServiceRead.VersionedLayerClient;
+    const mockedHRN = dataServiceRead.HRN.fromString(
+        "hrn:here:data:::mocked-hrn"
+    );
+    const mockedLayerId = "mocked-layed-id";
+    const fakeURL = "http://fake-base.url";
+
+    const testVersion = 42;
+
+    before(() => {
+        sandbox = sinon.createSandbox();
+        const settings = sandbox.createStubInstance(
+            dataServiceRead.OlpClientSettings
+        );
+
+        const versionedLayerClientParams = {
+            catalogHrn: mockedHRN,
+            layerId: mockedLayerId,
+            settings: (settings as unknown) as dataServiceRead.OlpClientSettings,
+            version: 5
+        };
+
+        versionedLayerClientNew = new dataServiceRead.VersionedLayerClient(
+            versionedLayerClientParams
+        );
+    });
+
+    beforeEach(() => {
+        getBlobStub = sandbox.stub(BlobApi, "getBlob");
+        getPartitionsStub = sandbox.stub(MetadataApi, "getPartitions");
+        getPartitionsByIdStub = sandbox.stub(QueryApi, "getPartitionsById");
+        getQuadTreeIndexStub = sandbox.stub(QueryApi, "quadTreeIndex");
+        getVersionStub = sandbox.stub(MetadataApi, "latestVersion");
+        getBaseUrlRequestStub = sandbox.stub(
+            dataServiceRead.RequestFactory,
+            "getBaseUrl"
+        );
+
+        getBaseUrlRequestStub.callsFake(() => Promise.resolve(fakeURL));
+    });
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    it("VersionedLayerClient instance should be initialized with VersionedLayerClientParams", async () => {

```

Comment:
Please test correct initialization of private properties

Reply:
`it("VersionedLayerClient instance should be initialized with VersionedLayerClientParams", async () => {
        assert.isDefined(versionedLayerClientNew);
        assert.equal(versionedLayerClientNew['version'], 5);
        assert.equal(versionedLayerClientNew['hrn'], "hrn:here:data:::mocked-hrn");
    });`

Reply:
fixed